test/common/http/http2/codec_impl_test.cc

```diff
@@ -782,8 +782,8 @@ TEST_P(Http2CodecImplTest, SmallMetadataVecTest) {
   // Generates a valid stream_id by sending a request header.
   TestRequestHeaderMapImpl request_headers;
   HttpTestUtility::addDefaultHeaders(request_headers);
-  EXPECT_CALL(request_decoder_, decodeHeaders_(_, true));
-  EXPECT_TRUE(request_encoder_->encodeHeaders(request_headers, true).ok());
+  EXPECT_CALL(request_decoder_, decodeHeaders_(_, false));
```

Comment:
I'm surprised these tests would end_stream at headers and then send metadata! 

docs and code do always handle header-only requests by removing end_stream=true on the headers and moving it to an empty data frame after the metadata, so I believe this is how it should be, and this was a minor error that just didn't impact unit tests.

Reply:
Yeah, quite strange, but should be fixed now.